app/src/main/java/com/checkin/app/checkin/auth/activities/AuthenticationActivity.kt

```diff
@@ -0,0 +1,287 @@
+package com.checkin.app.checkin.auth.activities
+
+import android.accounts.Account
+import android.accounts.AccountManager
+import android.content.Intent
+import android.os.Bundle
+import android.os.Handler
+import android.preference.PreferenceManager
+import android.util.Log
+import android.view.View
+import android.widget.FrameLayout
+import android.widget.TextView
+import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.net.toUri
+import androidx.lifecycle.Observer
+import androidx.navigation.NavController
+import androidx.navigation.findNavController
+import butterknife.BindView
+import butterknife.ButterKnife
+import com.checkin.app.checkin.R
+import com.checkin.app.checkin.auth.AuthFragmentInteraction
+import com.checkin.app.checkin.auth.AuthResultModel
+import com.checkin.app.checkin.auth.AuthViewModel
+import com.checkin.app.checkin.auth.exceptions.InvalidOTPException
+import com.checkin.app.checkin.auth.fragments.AuthOptionFragmentDirections
+import com.checkin.app.checkin.auth.fragments.AuthOtpFragment
+import com.checkin.app.checkin.auth.fragments.AuthOtpFragmentDirections
+import com.checkin.app.checkin.auth.services.DeviceTokenService
+import com.checkin.app.checkin.data.config.RemoteConfig
+import com.checkin.app.checkin.data.resource.Resource
+import com.checkin.app.checkin.home.activities.HomeActivity
+import com.checkin.app.checkin.user.models.UserModel
+import com.checkin.app.checkin.user.viewmodels.UserViewModel
+import com.checkin.app.checkin.utility.Constants
+import com.checkin.app.checkin.utility.Utils
+import com.checkin.app.checkin.utility.pass
+import com.checkin.app.checkin.utility.toast
+import com.facebook.login.LoginResult
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.tasks.Task
+import com.google.firebase.auth.*
+import java.util.*
+
+class AuthenticationActivity : AppCompatActivity(), AuthFragmentInteraction, AuthOtpFragment.AuthCallback {
+    @BindView(R.id.fl_circle_progress_container)
+    internal lateinit var flLoadingContainer: FrameLayout
+    @BindView(R.id.tv_auth_terms_conditions)
+    internal lateinit var tvTermsAndConditions: TextView
+
+    private val firebaseAuth: FirebaseAuth = FirebaseAuth.getInstance()
+    private val authViewModel: AuthViewModel by viewModels()
+    private val navController: NavController get() = findNavController(R.id.nav_host_authentication)
```

Comment:
Use lazy instead of overriding get()
Also, get() should always be in a new line